ts/screens/wallet/payment/ConfirmPaymentMethodScreen.tsx

```diff
@@ -257,119 +275,228 @@ const ConfirmPaymentMethodScreen: React.FC<Props> = (props: Props) => {
     );
   };
 
+  // Handle the PSP change, this will trigger
+  // a different callback for a payment with PayPal.
+  const handleChangePsp = isPayingWithPaypal
+    ? handleOnEditPaypalPsp
+    : props.pickPsp;
+
   const formData = props.payStartWebviewPayload
     .map<Record<string, string | number>>(payload => ({
       ...payload,
       ...getLookUpIdPO()
     }))
     .getOrElse({});
+
   const paymentMethod = props.getPaymentMethodById(wallet.idWallet);
+
   const ispaymentMethodCreditCard =
     paymentMethod !== undefined && isCreditCard(paymentMethod);
 
-  const showFeeContextualHelp = typeof fee !== "undefined" && fee > 0;
+  const formattedSingleAmount = formatNumberCentsToAmount(
+    verifica.importoSingoloVersamento,
+    true
+  );
+  const formattedTotal = formatNumberCentsToAmount(totalAmount, true);
+  const formattedFees = formatNumberCentsToAmount(fee ?? 0, true);
+
+  // Retrieve all the informations needed by the
+  // user interface based on the payment method
+  // selected by the user.
+  const paymentMethodInfo = getPaymentMethodInfo(
+    paymentMethod,
+    props.isPaypalEnabled
+  );
+
+  const paymentMethodSubject = paymentMethodInfo
+    .map(({ subject }) => subject)
+    .getOrElse("");
+
+  const paymentMethodExpiration = paymentMethodInfo
+    .map(({ expiration }) => expiration)
+    .getOrElse("");
+
+  const paymentMethodCaption = paymentMethodInfo
+    .map(({ caption }) => caption)
+    .getOrElse("");
+
+  const paymentMethodLogo = paymentMethodInfo
+    .map(({ logo }) => logo)
+    .getOrElse(<View />);
+
+  const paymentMethodA11yLabel = paymentMethodInfo
+    .map(({ accessibilityLabel }) => accessibilityLabel)
+    .getOrElse("");
```

Comment:
```suggestion
    const paymentMethodData = paymentMethodInfo.getOrElse({
    subject: "",
    expiration: "",
    caption: "",
    logo: <View />,
    accessibilityLabel: ""
  });
```

Instead of checking multiple times if `paymentMethodInfo` is `none` or `some`, what do you think about this refactoring? 
The logic is the same but shorter, with less code instructions

Reply:
Definitely better 😆 should be fixed in: https://github.com/pagopa/io-app/pull/3840/commits/ece66d1c44f640cdfffc8e3da238521459892bdc